File: python/Pc.py
#Saya Mohammad Ray Mosaid mengerjakan Latihan 3 dalam mata kuliah DPBO
#untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
#Aamiin.

#memanggil class yang sudah dibuat
from Processor import Processor
from Disk import Disk
from Ram import Ram

#membuat class Pc
class Pc:
    #membuat variabel class untuk menampung isi class (composition)
    __prodata = Processor(None, None)
    __diskdata = Disk(None, None, None)
    __ramdata = Ram(None, None)
    #membuat variabel class processor
    __totalprice = 0

    #membuat konstruktor
    def __init__(self, prodata, diskdata, ramdata):
        self.__prodata = prodata
        self.__diskdata = diskdata
        self.__ramdata = ramdata
        self.__totalprice = self.__prodata.getproprice() + self.__diskdata.getdiskprice() + self.__ramdata.getramprice()
    
    #membuat fungsi set untuk mengisi atribut private
    def setprodata(self, pcdata):
        self.__pcdata = pcdata
    def setdiskdata(self, diskdata):
        self.__diskdata = diskdata
    def setramdata(self, ramdata):
        self.__ramdata = ramdata
    def settotalprice(self, totalprice):
        self.__totalprice = self.__prodata.getproprice() + self.__diskdata.getdiskprice() + self.__ramdata.getramprice()
    
    #membuat fungsi get untuk me-return atribut private
    def getprodata(self):
        return self.__pcdata
    def getdiskdata(self):
        return self.__diskdata
    def getramdata(self):
        return self.__ramdata
    def gettotalprice(self):
        return self.__totalprice

    #membuat prosedur print data pc
    def printpcdata(self):
        self.__prodata.printprodata() #print data processor dengan memanggil prosedur print yang ada di class processor
        self.__diskdata.printdiskdata() #print data disk dengan memanggil prosedur print yang ada di class disk
        self.__ramdata.printramdata() #print data ram dengan memanggil prosedur print yang ada di class ram
        print("=========================")
        print("Total price: $", self.__totalprice)

File: c++/Pc.cpp
/*
    Saya Mohammad Ray Mosaid mengerjakan Latihan 3 dalam mata kuliah DPBO
    untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
    Aamiin.
*/
//memanggil class yang sudah dibuat
#include "Processor.cpp"
#include "Disk.cpp"
#include "Ram.cpp"

//memanggil header yang dibutuhkan
#include <iostream>
#include <string.h>
using namespace std;

//membuat class Pc
class Pc{
    private:
        //membuat variabel class untuk menampung isi class (composition)
        Processor prodata;
        Disk diskdata;
        Ram ramdata;
        //membuat variabel class processor
        int totalprice;

    public:
        //membuat konstruktor
        Pc(Processor p, Disk d, Ram r){
            this->prodata = p;
            this->diskdata = d;
            this->ramdata = r;
            this->totalprice = this->prodata.getproprice() + this->diskdata.getdiskprice() + this->ramdata.getramprice();
        }

        //membuat konstruktor dan juga prosedur set
        Pc(){
        }
        void setprodata(Processor p){
            this->prodata = p;
        }
        void setdiskdata(Disk d){
            this->diskdata = d;
        }
        void setramdata(Ram r){
            this->ramdata = r;
        }
        void settotalprice(){
            this->totalprice = this->prodata.getproprice() + this->diskdata.getdiskprice() + this->ramdata.getramprice();
        }

        //membuat fungsi get untuk me-return atribut private
        Processor getprodata(){
            return this->prodata;
        }
        Disk gediskdata(){
            return this->diskdata;
        }
        Ram getramdata(){
            return this->ramdata;
        }
        int gettotalprice(){
            return this->totalprice;
        }

        //membuat prosedur print data pc
        void printpcdata(){
            this->prodata.printprodata(); //print data processor dengan memanggil prosedur print yang ada di class processor
            this->diskdata.printdiskdata(); //print data disk dengan memanggil prosedur print yang ada di class disk
            this->ramdata.printramdata(); //print data ram dengan memanggil prosedur print yang ada di class ram
            cout << "=========================" << endl;
            cout << "Total price: $" << this->totalprice << endl;
        }

        ~Pc(){
        }
};
